### ISSUE TITLE
M4: BlueVela GOLD recipes — 2-node smoke + off-policy reference

### ISSUE THREAD

**Issue by cmadam** [cmadam/granite.build#60]
Part of #61.

## Goal

Runnable BlueVela recipes for the `gold-distill` step (#59): a minutes-scale smoke build for
plumbing, and the reference off-policy runs.

## The reference pair

Student **`granite-4.1-3b-base-hub`** (from `$KD_CODE_DIR/student_overlays/`) distilled from teacher
**`granite-4.2-30b`** (from `$KD_CODE_DIR/teacher_overlays/`), on
`data/en_sft_4.1/subsampled_0.4_shuffled_nothink.jsonl`. The teacher is frozen and forward-only
inside the training loop.

The dataset must be think-filtered (`*_nothink.jsonl`) — an assistant completion containing inline
`<think>...</think>` breaks gold's completion extraction.

## Recipes — `recipes/granite4-gold/lsf/`

Each dir is `build.yaml` + `parameters.yaml`, plus a `RUNBOOK.md` for the reference, mirroring
`recipes/granite4-350m/lsf/sft-10k-test/`:

- **`gold-smoke/`** — 2 nodes, small teacher/student pair, tiny dataset, `max_length: 2048`,
  `max_completion_length: 512`, `save_steps: 10`, `logging_steps: 1`, `dataset_num_proc: 8`,
  `poll_interval_seconds: 30`. Must finish in minutes: this is the plumbing test, run repeatedly.
- **`gold-offpolicy-2node/`** — the milestone config: `gradient_accumulation_steps: 12`, everything
  else as the reference.
- **`gold-offpolicy-4node/`** — the reference run: `gradient_accumulation_steps: 6`,
  `max_length: 16384`, full dataset, `lmbda: 0.0`, `vllm_num_servers: 0`.

Keep **one shared `build.yaml`** differing only by `parameters.yaml`, via the `$${VAR}` +
`--parameters-path` convention that `sft-10k-test/build.yaml` already uses. Follow its
`uri: "env://{{ binding.path }}"` output form.

Record the invariant as a comment in each `parameters.yaml`:

```
NUM_NODES * 8 * PER_DEVICE_BATCH_SIZE * GRAD_ACCUM == 192
```

Effective batch is what the LR, warmup ratio and cosine schedule were tuned against, so halving the
node count without doubling grad-accum changes the optimization, not just the throughput.

Set `compute_config.num_nodes` and `num_gpus_per_node: 8`, with
`launcher_config.resources.accelerators` (**per node**), `cluster`, and `zone` (= LSF queue).

Also keep `save_total_limit` high (20). An early, less-forgotten checkpoint is often the best one to
eval; a small limit deletes the sweet spot irrecoverably.

## Step build tests

`steps/gold-distill/skypilot/test-data/lsf/{build.yaml,buildtest.yaml}` with
`test/lsf/test_skypilot_lsf_gold_distill.py`, using the smoke parameters so Mode-2 is runnable.
`make publish-step` copies these into `test/steps/...` and `test-data/steps/...`, repointing
`space_uri`.

## Verification — live BlueVela via gbmcp

`build_start` -> `build_status` -> `build_job_log`, in this order:

1. Smoke GOLD, 2 nodes, tiny models. Read the rank-0 `cat` of the rendered config in the job log.
   Pin the monitor's progress regex here.
2. Reference **4-node** (`grad_accum: 6`, `max_length: 16384`) — the `EXEC_HOST`/blaunch scale test.
3. Reference **2-node** (`grad_accum: 12`) — the memory test.

Run 4-node **before** 2-node deliberately. The open risk is the ZeRO-3 memory ceiling: 16 ranks
instead of 32 doubles the per-GPU parameter/optimizer shard, and `per_device_train_batch_size` is
already 1 with gradient checkpointing already on, so there is no headroom lever left except
`max_length`. With a known-good 4-node baseline, a 2-node failure is unambiguously a capacity result
rather than a correctness bug. Mitigation ladder if it OOMs: more nodes (adjust `grad_accum`, keep
the 192 invariant) -> `max_length` 8192 -> offload the teacher.

Also add `NCCL_DEBUG=WARN` to the smoke `envs` and confirm `NET/IB` rather than `NET/Socket` in the
log — a TCP fallback is a run that "works" at a fraction of throughput, with nothing failing.

## Not in scope

On-policy (sampled-OPD) recipes. On-policy continues from a good off-policy checkpoint, so it cannot
be configured until #59's off-policy runs have produced one.
